dev-docs/RFCs/v8.0/luma-refactor.md

```diff
@@ -0,0 +1,83 @@
+# RFC: Refactoring luma.gl
+
+* **Author**: Tarek Sherif
+* **Date**: September, 2019
+* **Status**: **Draft**
+
+
+## Summary
+
+This RFC proposes steps to simplify luma.gl's API and ease its adoption among users.
+
+
+## Background
+
+luma.gl is currently a very large library relative to the functionality it provides and is difficult to approach for new users. This is due to several factors, among them the following:
+- 14 separate modules with no clear sense of how they're meant to interoperate
+- 201 exported classes and functions, again without a clear sense of their relationships
+- A large code base. ~733kB of source files - compare to e.g. a full 3D engine like three.js at ~1.2MB (1.6x), a low-level library like PicoGL at ~183kB (0.25x)
+- Several unfinished or unused APIs that make it hard to tell what's reliable
+- Documentation is sprawling and not focused on the API. Examples demonstrate random scenes rather than API features
+
+Clearly defining luma.gl's goals as a framework, and structuring the codebase and documentation around those goals, would benefit it in the following ways:
+- Reduce its complexity and associated maintenance costs
+- Reduce the weight it imposes on applications that use it
+- Make its value proposition clear to potential users
+- Simplify adoption
+
+## Overview
+
+The first step towards simplifying luma.gl is to be explicit about what its goals are. I propose that changes be made around the following key principles:
+- Targeting intermediate to advanced WebGL users
+- All systems implemented should solve explicitly identified problems. Experiments are encouraged, but those that fail to find a use case should be removed
+- Structured around levels of abstraction, primarily helper tools for low-level WebGL work, WebGL wrapper classes, and higher-level engine features
+- Increased focus on architectural fundamentals, simplicity, building systems that are easy to reason about. Less focus on using the latest APIs or frameworks
+
```

Comment:
Worth having a statement about glTF support ambitions? In its current form we have partial support exposed via a very "raw"/ user unfriendly API.

Reply:
I'd like to clean up glTF support, but I'm thinking we keep the scope of this RFC to cleanup that doesn't change any of the APIs we'll keep.